### PR TITLE
Eliminate unnecessary bitly calls in pi-apps client to improve accurate user-counts

### PR DESCRIPTION
**Pull request by Botspot**
With this PR, the bitly link is clicked ONLY if the app is successfully installed/uninstalled, *and* if it's not being reinstalled during an update.

### CODE REVIEW

**Comment by theofficialgman** [comment]
@Botspot this won't work as is because only the uninstall script gets passed `update` from the updater

**Comment by theofficialgman** [comment]
actually I have a solution... ugly but it works

**Comment by Botspot** [comment]
I found a much cleaner solution.
The `manage` process can directly determine if it is being run by `updater`, without relying on any flags or environment variables.

All `manage` needs to do is to check if `$(cat /proc/$PPID/comm)` == `updater`.

Source: https://askubuntu.com/a/1012236

**Comment by theofficialgman** [comment]
> All `manage` needs to do is to check if `$(cat /proc/$PPID/comm)` == `updater`.

I tested this out locally. it works. HOWEVER, it only works if we make sure to NEVER remove `#!/bin/bash` from the top of the `updater` script. otherwise it just returns `bash` and not `updater`

I know this because I tested locally with a script I wrote and only got `bash` to return, until I added the shebang at the top

**Comment by Botspot** [comment]
> I tested this out locally. it works. HOWEVER, it only works if we make sure to NEVER remove `#!/bin/bash` from the top of the `updater` script. otherwise it just returns `bash` and not `updater`

Interesting caveat. Just in case PPID somehow breaks on a marginal setup, I've also made it skip bitly_link if $3 is 'update'. Yes, I understand this won't have any immediate effect as `updater` doesn't pass the update flag, but it might as well be added now, just in case anything ever breaks PPID.

@theofficialgman, do you think anything else should be changed before merge?

**Comment by theofficialgman** [comment]
> @theofficialgman, do you think anything else should be changed before merge?

yes, I do. your commit is incorrect. it should be not equals

File: manage [hunk 1]
```diff
@@ -611,7 +611,7 @@ elif [ "$1" == 'install' ] || [ "$1" == 'uninstall' ];then
   status "${action^}ing \e[1m${app}\e[0m\e[96m..." | tee -a "$logfile"
   echo
   cd $HOME
-  if [[ "$3" == "update" ]]; then
+  if [ "$3" == "update" ]; then
```
Comment (theofficialgman) [inline manage:614]: enlighten me. what is the difference between the [[ ]] and [ ]. I know the former is more versatile but thats about it. if you have some resource that explains (well) that would be nice

**Comment by Botspot** [comment]
> actually @Botspot you can't add that check at all (even though your check is backwards currently).... in the case before the new updater is running. there will be no `update` for uninstall, and there will be `update` for install. this means on update you will get a bitly link click on install and not on uninstall due to the ||

That would only be true if the PPID conditional was not there.

**Comment by theofficialgman** [comment]
you must add a 4th arg as I did here if you want to have an actual check https://github.com/Botspot/pi-apps/pull/1999/commits/c2d11b5640ad06d1966e4bc430068050dcfe50ab . otherwise you break all bitly link clicks which occur during this update (which will likely be a lot)

the 4th arg can not fail

File: manage (hunk 1 above)
Comment (Botspot) [reply manage:614]: I don't have a particularly good source on-hand, but IIRC, the only time double brackets are necessary is for advanced operations like partial matches, regex, `=~`, things like that. Simple `==` or `-gt` don't need double brackets.

Comment (Jai-JAP) [reply manage:614]: https://unix.stackexchange.com/questions/306111/what-is-the-difference-between-the-bash-operators-vs-vs-vs

Comment (Jai-JAP) [reply manage:614]: @theofficialgman 
> https://unix.stackexchange.com/questions/306111/what-is-the-difference-between-the-bash-operators-vs-vs-vs

**Comment by theofficialgman** [comment]
> That would only be true if the PPID conditional was not there.

incorrect... your check was still backwards. it will click on update uninstall. and even if you reversed it to be correct, it would then click on update install

**Comment by Botspot** [comment]
> there will be no `update` for uninstall, and there will be `update` for install. this means on update you will get a bitly link click on install and not on uninstall due to the ||

At least we're both getting things backwards! (Actually there will be no `update` for _install_, and there will be `update` for uninstall)
You're right about the || - it should have been `&&` with `[ $3 != update ]`

**Comment by theofficialgman** [comment]
no I'm right.. read again your commit. 
I will explain it to you

the updater (before the user restarts after this PR merges) will still call apps like so
for uninstall it passes the update as $3:
```
"${DIRECTORY}/manage" uninstall "$app" update
```
for install, it does NOT pass update as $3 since this didn't exist in the loaded script
```
"${DIRECTORY}/manage" install "$app"
```

here is your commit:
```
if [ $(cat /proc/$PPID/comm) != updater ] || [ "$3" == "update" ];then
  bitly_link "$app" "$action" &
fi
```

assuming $$PID does come out to be updater:

your check for update uninstall will pass, since $3 == update. meaning the bitly link gets clicked, BAD

your check for update install will NOT pass, since $3 does not contain a value, meaning the bitly link does NOT get clicked

notice how in my comment, I said
>  it will click on update uninstall. and even if you reversed it to be correct, it would then click on update install

this is correct with my explanation above

**Comment by theofficialgman** [comment]
> You're right about the || - it should have been && with [ $3 != update ]

anyway, this is fine. with how the script is now, the PPID should function correctly and keep the bitly link from getting clicked. In the future if the PPID check gets messed up, the lack of "update" will allow the bitly link to get clicked

**Comment by theofficialgman** [comment]
> This picture, particularly line 2, seems especially relevant for this PR: 

well said. ignore my argumentative attitude right now. forgive me after watching a that marvel movie which literally had demon summoning in it 😆

**Pull request merged**